evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+import pandas as pd
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils import import_or_raise
+
+
+class ARIMARegressor(Estimator):
+    """
+    Autoregressive Integrated Moving Average Model.
+    The three parameters (p, d, q) are the AR order, the degree of differencing, and the MA order.
+    More information here: https://www.statsmodels.org/devel/generated/statsmodels.tsa.arima_model.ARIMA.html
+
+    """
+    name = "ARIMA Regressor"
+    hyperparameter_ranges = {
+        "p": Integer(0, 10),
+        "d": Integer(0, 10),
+        "q": Integer(0, 10),
+    }
+    model_family = ModelFamily.ARIMA
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, date_column=None, trend='n', p=1, d=0, q=0,
+                 random_seed=0, **kwargs):
+        order = (p, d, q)
+        parameters = {'order': order,
+                      'trend': trend}
+
+        parameters.update(kwargs)
+        self.date_column = date_column
+
+        p_error_msg = "ARIMA is not installed. Please install using `pip install statsmodels`."
+
+        arima = import_or_raise("statsmodels.tsa.arima.model", error_msg=p_error_msg)
+        try:
+            arima.ARIMA(endog=np.zeros(p + d + q + 1), **parameters)
+        except TypeError:
+            raise TypeError("Unable to instantiate ARIMA due to an unexpected argument")
+        parameters.update({'p': p,
+                           'd': d,
+                           'q': q})
+
+        super().__init__(parameters=parameters,
+                         component_obj=None,
+                         random_seed=random_seed)
+
+    def _get_dates_fit(self, X, y):
+        date_col = None
+
+        if isinstance(y.index, pd.DatetimeIndex):
+            date_col = y.index
+        if X is not None:
+            if self.date_column in X.columns:
+                date_col = X.pop(self.date_column)
+            elif isinstance(X.index, pd.DatetimeIndex):
+                date_col = X.index
+
+        if date_col is None:
+            msg = "ARIMA regressor requires input data X to have a datetime column specified by the 'date_column' parameter. " \
+                  "If not it will look for the datetime column in the index of X or y."
```

Comment:
To confirm this error message vs the one below, this will look for the datetime in "the index of X or y", but predict will only look at X? :)

Reply:
Well technically because the `predict` function wasn't going to use `y` I changed that, but since it does check `y` for a datetime index, I'll change it back again, thanks!